test/e2e/on-request-error/basic/basic.test.ts

```diff
@@ -40,21 +46,38 @@ describe('on-request-error - basic', () => {
     expect(payload.message).toBe(name)
     expect(count).toBe(1)
 
-    validateRequestByName(payload.request, url, isMiddleware)
+    validateRequestByName({
+      payload: payload,
+      url,
+      isMiddleware,
+      renderSource,
+    })
   }
 
-  function validateRequestByName(
-    request: any,
-    url: string,
-    isMiddleware: boolean = false
-  ) {
+  function validateRequestByName({
+    payload,
+    url,
+    renderSource,
+    isMiddleware = false,
+  }: {
+    payload: any
+    url: string
+    renderSource?: string
+    isMiddleware: boolean
+  }) {
+    const { request } = payload
     if (isMiddleware) {
       // For middleware, the URL is absolute url with host
       expect(request.url).toMatch(/^http:\/\//)
       expect(request.url).toMatch(url)
     } else {
       expect(request.url).toBe(url)
     }
+
+    if (renderSource) {
```

Comment:
I think it'd make sense to always verify `renderSource`, even when explicitly verifying if it's undefined to ensure we don't end up in a case where we aren't checking a `renderSource` value (because we weren't expecting one) but there is one present.